var/spack/repos/builtin/packages/rocm-validation-suite/package.py

```diff
@@ -183,10 +183,14 @@ def patch(self):
             filter_file(
                 "@ROCM_PATH@/rvs", self.spec.prefix.rvs, "rvs/conf/deviceid.sh.in", string=True
             )
-        elif "@5.2.0:" in self.spec:
+        elif "@5.2.0:5.4" in self.spec:
```

Comment:
Version (range) checks should be using `self.spec.satisfies()`.

See line 204 and the docs at https://spack.readthedocs.io/en/latest/packaging_guide.html#testing-spec-constraints.

Reply:
Also, I think the range `"@5.2:5.4" would suffice.

Reply:
I also suggest fixing the check at line 182 too.

Reply:
Updated as recommended.